### ISSUE TITLE
Error: Cannot find module 'typescript-logging' on Root Server (Debian 8.5)

### ISSUE THREAD

**Issue by messerbill** [mreuvers/typescript-logging#22]
Hi,
locally my typescript-logging works without any problems. But on my  Root Server the following message appears on Server Startup:

`Error: Cannot find module 'typescript-logging'
    at Function.Module._resolveFilename (module.js:527:15)
    at Function.Module._load (module.js:476:23)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/root/controllers/logger.ts:1:1)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)`

This only happens with typescript-logging. All other modules are loaded corrrectly. 

the Import:

> import {Category,CategoryLogger,CategoryServiceFactory,CategoryDefaultConfiguration,LogLevel} from "typescript-logging"

What i tried so far:

 - Removing all node_modules and re-install them.
 - Installing typescript-logging locally
 - Installing typescript-logging globally
 - Installing typescript-logging locally & globally
 - change import Path in the Typescript file

Nothing worked.....on the Root Server. But it always works locally.... What am i doing wrong here?

**Comment by mreuvers** [comment]
Hello @messerbill 

I am not sure what you mean with "root server" here (aside it being a debian distribution). What typescript-logging version do you use? npm/node versions (local/server)?

What do you start exactly on the server (what/how)? Is everything bundled by e.g. webpack? Is the bundling different. is the setup different (local vs server)?

That aside, typescript-logging should be part of the normal dependencies in package.json and be bundled along. It could be differences in npm/node versions as well.

So in short there's not much to go on here. Do you think you could make a simple reproduction case so I can see it fail myself, as right now I am stumbling in the dark.

**Comment by messerbill** [comment]
hi @mreuvers ,
thanks for your reply. "Root Server" is my webserver / live server in the internet. 

I am using node version `v8.6.0` and NPM version `5.4.2`. Typescript-logging is part of my normal dependencies in my package.json: `"typescript-logging": "^0.4.0"`

And my Implementation:

```
import {Category,CategoryLogger,CategoryServiceFactory,CategoryDefaultConfiguration,LogLevel} from "typescript-logging"

/**
 * Logger for the Project
 * @module
 */
export module Logger {

    export const catGame = new Category("game")
    //export const catProd = new Category("product", catRoot)
    export const catConnection = new Category("connection")
    export const catMat = new Category("MAT")

    export const gameLog: CategoryLogger = CategoryServiceFactory.getLogger(catGame)
    export const conLog: CategoryLogger = CategoryServiceFactory.getLogger(catConnection)
    export const matLog: CategoryLogger = CategoryServiceFactory.getLogger(catMat)


    /**
     * sets the logger defaults to LogLevel.info
     * @export
     */
    export function setLoggerDefaults() {
        CategoryServiceFactory.setDefaultConfiguration(new CategoryDefaultConfiguration(LogLevel.Info))
    }
    
}
```

**Comment by eluft84** [comment]
I have the same issue running on node v6.9.0 and npm 3.10.8

**Comment by mreuvers** [comment]
Hello @eluft84 and @messerbill 

The released distribution is broken, I reproduced it now 
(not sure how this even happened, maybe an oops on my side).

Could you please go back to 0.4.0 explicitly for now? Make sure to remove the ^ in your package.json and lock it to 0.4.0 as workaround. 
Also please make sure to drop the existing typescript-logging e.g. from node_modules first, then reinstall 0.4.0. This should work.

I will try to get a proper version out soon.

**Comment by mreuvers** [comment]
Hello @eluft84  and @messerbill 

I just published 0.4.2 to npm, which is a valid distribution again. Tested it against node and it works properly again.

I leave this bug open for a few more days for you to test, could you please confirm 0.4.2 works for you as well? Thank you.

**Comment by messerbill** [comment]
this works! Thank you for the fast update <3

**Comment by mreuvers** [comment]
Glad to hear it did :)

**Comment by eluft84** [comment]
I can only agree - it works! Thanks a lot for the fast update.

**Comment by mreuvers** [comment]
Good to hear it works for both of you! :) Closing this bug now. Have fun coding!
